crates/provider-sdk/src/provider.rs

```diff
@@ -554,15 +554,16 @@ async fn delete_link_for_provider<P>(
 where
     P: Provider,
 {
-    if ld.source_id == *connection.provider_id {
-        if let Err(e) = provider.delete_link_as_source(&ld.target).await {
+    if *ld.source_id == *connection.provider_id {
+        if let Err(e) = provider.delete_configured_link_as_source(&ld).await {
             error!(error = %e, target = &ld.target, "failed to delete link to component");
         }
-    } else if ld.target == *connection.provider_id {
-        if let Err(e) = provider.delete_link_as_target(&ld.source_id).await {
+    } else if *ld.target == *connection.provider_id {
+        if let Err(e) = provider.delete_configured_link_as_target(&ld).await {
             error!(error = %e, source = &ld.source_id, "failed to delete link from component");
         }
     }
+    warn!("delete_link() is deprecated, consider using delete_configured_link_*() variants. In a future release delete_link() will be removed.");
```

Comment:
Won't we still need to call the `delete_link_as_*` methods in order to not make this a breaking change?

Reply:
So I think when I made this, I wanted the change to be *breaking* in Provider SDK -- and get it out as fast as possible. Since we're not at 1.x.x, `0.6.0` would have been a breaking change (now it needs to be like `0.8.0` I think

Reply:
The warning should probably just not be there actually, I can remove it